src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,277 @@
+                newEvent.addVehicleRequirementId(stoi(entry.at(6)));
+
+                std::cout << "Adding Event #" << entry.at(0) << " to pending queue" << std::endl;
+
+                pendingQueue->push_back(newEvent);
+            }
+        }
+    }
+
+    void DistrictResources::processVehicleRequirementsSampleData(const std::string &fileName)
+    {
+        std::vector<std::vector<std::string>> reqts;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            reqts.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : reqts)
+        {
+            if (header)
+            {
+                header = false;
+                continue;
+            }
+            uint16_t vehicleReqtId = stoi(entry.at(0));
+
+            if (entry.at(1) == "FireEngine")
+            {
+                FireEngine newReqt = FireEngine();
+                newReqt.setCurWaterVolumeLitres(stof(entry.at(2)));
+                newReqt.setCurNumCrew(stoi(entry.at(3)));
+                if (vehicleRequirements.find(vehicleReqtId) == vehicleRequirements.end())
+                {
+                    vehicleRequirements[vehicleReqtId] = *new std::vector<Vehicle>;
+                }
+                std::cout << "Adding vehicle with Requirement ID: #" << vehicleReqtId << " to global list" << std::endl;
+                vehicleRequirements[vehicleReqtId].push_back(newReqt);
+            }
+            else if (entry.at(1) == "FireLadder")
+            {
+                FireLadder newReqt = FireLadder();
+                newReqt.setCurWaterVolumeLitres(stof(entry.at(2)));
+                newReqt.setCurNumCrew(stoi(entry.at(3)));
+                if (vehicleRequirements.find(vehicleReqtId) == vehicleRequirements.end())
+                {
+                    vehicleRequirements[vehicleReqtId] = *new std::vector<Vehicle>;
+                }
+
+                std::cout << "Adding vehicle with Requirement ID: #" << vehicleReqtId << " to global list" << std::endl;
+                vehicleRequirements[vehicleReqtId].push_back(newReqt);
+            }
+        }
+    }
+
+    std::vector<Vehicle> *DistrictResources::getVehicleRequirements(uint32_t eventID) {
+        return &vehicleRequirements[eventID];
+    }
+
+    double calculateDistance(Location *locationA, Location *locationB)
+    {
+        float xDist = locationA->getXCoord() - locationB->getXCoord();
+        float yDist = locationA->getYCoord() - locationB->getYCoord();
+        double xDistSquared = std::pow(xDist, 2.0);
+        double yDistSquared = std::pow(yDist, 2.0);
+        return std::sqrt(xDistSquared + yDistSquared);
+    }
```

Comment:
```suggestion
    double calculateDistance(const Location & locationA, const Location & locationB)
    {
        float xDist = locationA.getXCoord() - locationB.getXCoord();
        float yDist = locationA.getYCoord() - locationB.getYCoord();
        double xDistSquared = std::pow(xDist, 2.0);
        double yDistSquared = std::pow(yDist, 2.0);
        return std::sqrt(xDistSquared + yDistSquared);
    }
```
This code is probably safer, because you don't need the address for the logic in this code and it will not accept a null value as an input parameter (which, with pointers, would cause a seg fault).

Note that to accept const-refs, you will probably need to overload your `getXCoord()` and `getXCoord()` getters to also provide a const version (unless it currently returns a value instead of a reference).